src/MyEventEmitter.js

```diff
@@ -1,12 +1,70 @@
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+class MyEventEmitter {
+  constructor() {
+    this.listeners = [];
+  }
+
+  on(name, callback) {
+    if (
+      typeof callback === 'function'
+      && typeof name === 'string'
+    ) {
+      this.listeners.push({
+        name, callback,
+      });
+    }
+  }
+
+  once(eventName, callback) {
+    function onceFn(data) {
+      this.off(eventName, callback);
+      callback(data);
+    }
+    this.on(eventName, onceFn);
+  }
+
+  off(eventName, callback) {
+    this.listeners = this.listeners.filter(
+      listener => !(listener.name === eventName
+        && listener.callback === callback)
+    );
+  }
+
+  emit(eventName, data) {
+    this.listeners
+      .filter(({ name }) => name === eventName)
+      .forEach(
+        ({ callback }) => callback.apply(this, [this, ...data]));
+  }
+  prependListener(name, callback) {
+    if (
+      typeof callback === 'function'
+      && typeof name === 'string'
+    ) {
+      this.listeners.unshift({
+        name, callback,
+      });
+    }
+  }
+
+  prependOnceListener(eventName, callback) {
+    function onceFn(data) {
+      this.off(eventName, callback);
+      callback(data);
+    }
```

Comment:
this func is reusable, consider storing it in separate const, or refactor your code so that you don't implement same logic in prependOnceListener and once